Code/Tools/AssetProcessor/native/utilities/BuilderManager.inl

```diff
@@ -19,6 +19,29 @@ namespace AssetProcessor
         TNetResponse netResponse;
         netRequest.m_request = request;
 
+        struct BuildTracker final
+        {
+            BuildTracker(const Builder& builder, const AZStd::string& sourceFile, const AZStd::string& task)
+                : m_builder(builder)
+                , m_sourceFile(sourceFile)
+                , m_task(task)
+            {
+                AZ_TracePrintf(AssetProcessor::ConsoleChannel, "Request started builder [%s] task (%s) %s \n",
+                    m_builder.UuidString().c_str(), m_task.c_str(), m_sourceFile.c_str());
+            }
+
+            ~BuildTracker()
+            {
+                AZ_TracePrintf(AssetProcessor::ConsoleChannel, "Request stopped builder [%s] task (%s) %s \n",
+                    m_builder.UuidString().c_str(), m_task.c_str(), m_sourceFile.c_str());
+            }
+
+            const Builder& m_builder;
+            const AZStd::string& m_sourceFile;
```

Comment:
Do you mean for these to hold references to objects, why not have local copies of the string / smart pointer for the builder?

Reply:
these were all meant to be references, there are no risks since the instance is function scoped and take in the argument parameters as referenced values